tests/test_analog.py

```diff
@@ -388,3 +390,27 @@ def apply_hamevo_and_compare_expected(psi, values):
     apply_hamevo_and_compare_expected(psi, values)
     assert len(hamevo._cache_hamiltonian_evo) == 2
     assert values_cache_key in previous_cache_keys
+
+
+@pytest.mark.parametrize(
+    "generator, time_param, result",
+    [
+        (RX(0, "x"), "x", True),
+        (RX(1, 0.5), "y", False),
+        (RX(0, "x"), "y", False),
+        (RX(0, "x"), torch.tensor(0.5), False),
+        (RX(0, torch.tensor(0.5)), torch.tensor(0.5), False),
```

Comment:
Technically `RX` should not be a generator, I don't think it is Hermitian. For the purpose of this test it's not so relevant, but still I would keep only cases using `Scale` and expand those a little bit.

Reply:
As discussed, I'll leave the `RX` so that test handles all edge cases.